src/main/java/org/cactoos/list/JoinedListIterator.java

```diff
@@ -47,7 +47,7 @@
     /**
      * {@link List} of {@link ListIterator}.
      */
-    private final List<ListIterator<T>> listiters;
+    private final List<ListIterator<? extends T>> listiters;
```

Comment:
@DmitryBarskov I think that should be `List<? extends ListIterator<...`, here and everywhere else :)